docs/notebook-demos/core-concepts.ipynb

```diff
@@ -0,0 +1,447 @@
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import geopyspark as gps"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Core Concepts\n",
+    "\n",
+    "Because GeoPySpark is a binding of an existing project, GeoTrellis, some terminology and data representations have carried over. This section seeks to explain this jargon in addition to describing how GeoTrellis types are represented in GeoPySpark."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Rasters\n",
+    "\n",
+    "GeoPySpark differs in how it represents rasters from other geo-spatial Python libraries like rasterIO. In GeoPySpark, they are represented by the `Tile` class. This class contains a numpy array (refered to as `cells`) that represents the cells of the raster in addition to other information regarding the data. Along with `cells`, `Tile` can also have the `no_data_value` of the raster.\n",
+    "\n",
+    "**Note**: All rasters in GeoPySpark are represented as having multiple bands, even if the original raster just contained one."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "arr = np.array([[[0, 0, 0, 0],\n",
+    "                 [1, 1, 1, 1],\n",
+    "                 [2, 2, 2, 2]]], dtype=np.int16)\n",
+    "\n",
+    "# The resulting Tile will set -10 has the no_data_value for the raster\n",
+    "gps.Tile.from_numpy_array(numpy_array=arr, no_data_value=-10)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# The resulting Tile will have no no_data_value\n",
+    "gps.Tile.from_numpy_array(numpy_array=arr)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Extent\n",
+    "\n",
+    "Describes the area on Earth a raster represents. This area is represented by coordinates that are in some Coordinate Reference System. Thus, depending on the system in use, the values that outline the `extent` can vary. `Extent` can also be refered to as a *bounding box*.\n",
+    "\n",
+    "**Note**: The values within the `Extent` must be `float`s and not `double`s."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "extent = gps.Extent(0.0, 0.0, 10.0, 10.0)\n",
+    "extent"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## ProjectedExtent\n",
+    "\n",
+    "`ProjectedExtent` describes both the area on Earth a raster represents in addition to its CRS. Either the EPSG code or a proj4 string can be used to indicate the CRS of the `ProjectedExtent`."
```

Comment:
Do you think that someone who is using GeoPySpark would already be familiar with these terms? At least from my experience, I was exposed to both of these terms pretty early on in my classes.